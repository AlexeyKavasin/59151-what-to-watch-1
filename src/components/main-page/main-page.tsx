import * as React from "react";
import {connect} from "react-redux";
import {Link} from 'react-router-dom';
import Catalog from "../catalog/catalog";
import {IApp} from "../../interfaces";
import {Footer} from "../footer/footer";
import {FullWidthPlayer} from "../full-width-player/full-width-player";

class MainPage extends React.PureComponent<IApp, null> {

  render() {
    const {filmOnTheMainPage, isAuthorized, userData, onSignInClick, isFullWidthPlayerActive, toggleFullWidthPlayer} = this.props;

    return <React.Fragment>
      {
        isFullWidthPlayerActive ?
        <FullWidthPlayer 
          videoSrc={filmOnTheMainPage.video_link}
          poster={filmOnTheMainPage.background_image}
          runTime={filmOnTheMainPage.run_time}
          toggleFullWidthPlayer={toggleFullWidthPlayer}
          filmName={filmOnTheMainPage.name}
        /> :
        <React.Fragment>
        <section className="movie-card">
        <div className="movie-card__bg">
          <img src={filmOnTheMainPage.background_image} alt={filmOnTheMainPage.name} />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <header className="page-header movie-card__head">
          <div className="logo">
            <Link className="logo__link" to="/">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </Link>
          </div>

          <div className="user-block">
            {isAuthorized && userData ? (
              <div className="user-block__avatar">
                <img src={`https://es31-server.appspot.com/${userData.avatar_url}`} alt={userData.name} width="63" height="63" />
              </div>
            ) :
              (<Link className="user-block__link" to="/login" onClick={onSignInClick}>Sign In</Link>)
            }
          </div>
        </header>

        <div className="movie-card__wrap">
          <div className="movie-card__info">
            <div className="movie-card__poster">
              <Link to={`/film/${filmOnTheMainPage.id}`}>
                <img src={filmOnTheMainPage.poster_image} alt={filmOnTheMainPage.name} width="218" height="327" />
              </Link>
            </div>

            <div className="movie-card__desc">
              <h2 className="movie-card__title">{filmOnTheMainPage.name}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{filmOnTheMainPage.genre}</span>
                <span className="movie-card__year">{filmOnTheMainPage.released}</span>
              </p>

              <div className="movie-card__buttons">
                <button onClick={toggleFullWidthPlayer} className="btn btn--play movie-card__button" type="button">
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s"></use>
                  </svg>
                  <span>Play</span>
                </button>
                <button className="btn btn--list movie-card__button" type="button">
                  <svg viewBox="0 0 19 20" width="19" height="20">
                    <use xlinkHref="#add"></use>
                  </svg>
                  <span>My list</span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </section>

      <div className="page-content">

        <Catalog/>

        <Footer/>

      </div>
      </React.Fragment>
      }
    </React.Fragment>;
  }
}

const mapStateToProps = (state, ownProps) => ({
  ownProps,
  filmOnTheMainPage: state[`DATA`].films[9]
});

export {MainPage};

export default connect(
    mapStateToProps,
    null
)(MainPage);
