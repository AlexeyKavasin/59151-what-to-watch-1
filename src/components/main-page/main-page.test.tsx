import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {HashRouter} from 'react-router-dom';
import {MainPage} from './main-page';

jest.mock(`../catalog/catalog`, () => () => `Catalog`);

describe(`MainPage snapshot tests`, () => {
  it(`MainPage renders correctly`, () => {
    const mock = jest.fn();
    const tree = renderer
      .create(
          <HashRouter>
            <MainPage
              isAuthorizationRequired={false}
              isAuthorized={false}
              onSignInClick={mock}
              userData={null}/>
          </HashRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
