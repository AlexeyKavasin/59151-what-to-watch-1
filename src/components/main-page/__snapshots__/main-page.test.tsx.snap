// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage snapshot tests MainPage renders correctly 1`] = `
Array [
  <section
    className="movie-card "
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Matrix"
        src="https://es31-server.appspot.com/wtw/static/film/preview/matrix.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="#/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="#/login"
          onClick={[Function]}
        >
          Sign In
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <a
            href="#/film/0"
            onClick={[Function]}
          >
            <img
              alt="Matrix"
              height="327"
              src="https://es31-server.appspot.com/wtw/static/film/poster/matrix.jpg"
              width="218"
            />
          </a>
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Matrix
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              1999
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[MockFunction]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <Catalog />
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="#/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
